sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorExecutor.java

```diff
@@ -175,14 +175,6 @@ private void scheduleCompletePendingTasks() {
     public void close() {
         if (!isDisposed.getAndSet(true)) {
             close(true, "ReactorExecutor.close() was called.");
-
-            try {
-                if (!disposeSemaphore.tryAcquire(timeout.toMillis(), TimeUnit.MILLISECONDS)) {
-                    logger.info("Unable to acquire dispose reactor semaphore within timeout.");
-                }
-            } catch (InterruptedException e) {
-                logger.warning("Could not acquire semaphore to finish close operation.", e);
-            }
```

Comment:
@conniey Why do we try to acquire the semaphore when closing? We are not doing anything after acquiring the semaphore.

Reply:
I think this could be removed. 

1. Also this `close()` will call `scheduleCompletePendingTasks()` which again try to acquire the semaphore. 
2. `scheduleCompletePendingTasks()` ": This function calls  
 reactor.stop();
 reactor.process();
 shouldn't this be called in reverse order i.e.  call `reactor.process()` first. ?

Reply:
the original reason I had added this is because I wanted the rest of the tasks in the task queue to complete before removing the executor.

Reply:
@conniey This could result in a deadlock as the underlying connection may be closed and the tasks in the queue will not complete and hold on to the semaphore. I profiled this and all the executor threads were stuck waiting to acquire the semaphore.